itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcrossInstances.java

```diff
@@ -1639,12 +1655,68 @@ public void testFailureUnsupportedAuthorizerReplication() throws Throwable {
         .run("insert into table1 values (2)");
     try {
       primary.dump(primaryDbName, clause);
+      Assert.fail();
     } catch (SemanticException e) {
       assertEquals("Invalid config error : Authorizer sentry not supported for replication  " +
         "for ranger service.", e.getMessage());
       assertEquals(ErrorMsg.REPL_INVALID_CONFIG_FOR_SERVICE.getErrorCode(),
         ErrorMsg.getErrorMsg(e.getMessage()).getErrorCode());
     }
+    //This is now non recoverable error
+    try {
+      primary.dump(primaryDbName, clause);
+      Assert.fail();
+    } catch (Exception e) {
+      assertEquals(ErrorMsg.REPL_FAILED_WITH_NON_RECOVERABLE_ERROR.getErrorCode(),
+        ErrorMsg.getErrorMsg(e.getMessage()).getErrorCode());
+    }
+    //Delete non recoverable marker to fix this
```

Comment:
Can you please also add a check to ensure that there is no marker file at this point of time

Reply:
Also, if you run a repl load now, it should also fail.

Reply:
I have added a check for marker file. But that will just test the fs delete call.
Also load will fail as the dump doesn't exist, not because of non recoverable error. Those kind of tests are already present and not part of this feature.

Reply:
Also load will not fail due to non recoverable error as the dump path itself is not there. At Replication Semantic analyzer level itself it will just return.